pkg/policy/repository.go

```diff
@@ -120,6 +121,110 @@ func (p *Repository) AllowsIngressLabelAccess(ctx *SearchContext) api.Decision {
 	return decision
 }
 
+func wildcardL3L4Rule(proto api.L4Proto, port int, endpoints api.EndpointSelectorSlice,
+	ruleLabels labels.LabelArray, l4Policy L4PolicyMap) {
+	for k, filter := range l4Policy {
+		if proto != filter.Protocol || (port != 0 && port != filter.Port) {
+			continue
+		}
+		switch filter.L7Parser {
+		case ParserTypeNone:
+			continue
+		case ParserTypeHTTP:
+			// Wildcard at L7 all the endpoints allowed at L3 or L4.
+			for _, sel := range endpoints {
+				filter.L7RulesPerEp[sel] = api.L7Rules{
+					HTTP: []api.PortRuleHTTP{{}},
+				}
+			}
+			filter.Endpoints = append(filter.Endpoints, endpoints...)
+			filter.DerivedFromRules = append(filter.DerivedFromRules, ruleLabels)
+			l4Policy[k] = filter
+		case ParserTypeKafka:
+			// Wildcard at L7 all the endpoints allowed at L3 or L4.
+			for _, sel := range endpoints {
+				rule := api.PortRuleKafka{}
+				rule.Sanitize()
```

Comment:
Isn't this a `nop` ? Why do we need a `rule.Sanitize()` on an empty rule. This will simply return a `nil` here. This is an unnecessary step IMO.

Reply:
It is necessary to set the apiKeyint field. This is the only way to do that.

Reply:
And that is necessary only for Kafka rules. There are no private fields in HTTP rules.

Reply:
apiKeyInt is only set on this situation

```
	if len(kr.APIKey) > 0 {
		n, ok := KafkaAPIKeyMap[strings.ToLower(kr.APIKey)]
		if !ok {
			return fmt.Errorf("invalid Kafka APIKey :%q", kr.APIKey)
		}
		kr.apiKeyInt = append(kr.apiKeyInt, n)
	}
```
from what I can see, this code is never hit in this situation.

Same for
```
if len(kr.APIVersion) > 0 {
		n, err := strconv.ParseInt(kr.APIVersion, 10, 16)
		if err != nil {
			return fmt.Errorf("invalid Kafka APIVersion :%q",
				kr.APIVersion)
		}
		n16 := int16(n)
		kr.apiVersionInt = &n16
	}
```
So none of the private variables are being filled as it is. What am I not understanding here?